docs/design/services/dynamodb/high-level-library/prototype/option-1/sync/Prototype.java

```diff
@@ -0,0 +1,1026 @@
+     *         public Instant getCreationTime() { return this.creationTime; }
+     *         public void setPartitionKey(String partitionKey) { this.partitionKey = partitionKey; }
+     *         public void setCreationTime(Instant creationTime) { this.creationTime = creationTime; }
+     *     }
+     *
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
+     *         Table table = client.getTable("my-table");
+     *
+     *         MyItem myItem = new MyItem();
+     *         myItem.setPartitionKey(UUID.randomUUID());
+     *         myItem.setCreationTime(Instant.now());
+     *
+     *         table.putObject(myItem);
+     *     }
+     * </code>
+     */
+    void putObject(Object item);
+
+    /**
+     * Invoke DynamoDB to create or override an Item in this table.
+     *
+     * This will convert the provided object into an {@link Item} automatically using the default Object-to-Item
+     * {@link ItemAttributeValueConverter}, unless an alternate converter has been overridden for the provided type.
+     *
+     * This method provides more options than {@link #putObject(Object)} like conditions or consumed capacity.
+     *
+     * Usage Example:
+     * <code>
+     *     public class MyItem {
+     *         @Attribute("partition-key")
+     *         @Index(AttributeIndexType.PARTITION_KEY)
+     *         private String partitionKey;
+     *
+     *         @Attribute
+     *         private int version;
+     *
+     *         public String getPartitionKey() { return this.partitionKey; }
+     *         public int getVersion() { return this.version; }
+     *         public void setPartitionKey(String partitionKey) { this.partitionKey = partitionKey; }
+     *         public void setVersion(int version) { this.version = version; }
+     *     }
+     *
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
+     *         Table table = client.getTable("my-table");
+     *
+     *         MyItem myItem = new MyItem();
+     *         myItem.setPartitionKey(UUID.randomUUID());
+     *         myItem.setVersion(2);
+     *
+     *         table.putObject(PutObjectRequest.builder(myItem)
+     *                                         .condition("version = :expected_version")
+     *                                         .putConditionAttribute(":expected_version", 1)
+     *                                         .build());
+     *     } catch (ConditionFailedException e) {
+     *         System.out.println("Precondition failed.");
+     *         throw e;
+     *     }
+     * </code>
+     */
+    <T> PutObjectResponse<T> putObject(PutObjectRequest<T> putRequest)
+            throws ConditionFailedException;
+
+    /**
+     * Invoke DynamoDB to retrieve an Item in this table, based on its partition key (and sort key, if the table has one).
+     *
+     * This method is optimized for performance, and provides no additional response data. For additional options
```

Comment:
Can you explain why this will have more performance? Even though we might return part of response, our marshallers stills need to marshall the entire response right

Reply:
I picture the marshalling chain to roughly work like this:
1. Create a context for the current item.
2. Find a converter for this context:
   a. Iterate over each `contextSatisfies` converter to see if it can marshal this context. If so, use that converter.
   b. Look up the context.attributeSchema().javaType() in a `typeConverterCache`. If a converter is found, use that converter.
   c. Repeat (b) for the parent class (eventually, the base Object converter will be used), caching the result as the converter for that child type.
3. Use the convert to transform the item.

2(b) will converge on O(1) 
2(a) will always be O(# of contextSatisfies marshallers)